### ISSUE TITLE
ManagedChannelBuilder doesn't have maxMessageSize

### ISSUE THREAD

**Issue by ejona86** [grpc/grpc-java#2307]
This makes it more annoying to override the maxMessageSize, because you have to know which implementation you are using.

Note that if you specify the actual implementation, this is easy, but the problem is needing to specify the implementation.

Instead of:

```
ManagedChannelBuilder.forTarget(...)
   .maxMessageSize(SIZE)
   .build();
```

You are forced to do:

```
NettyChannelBuilder.forTarget(...)
   .maxMessageSize(SIZE)
   .build();
```

**Comment by ejona86** [comment]
This is slightly tricky since things like InProcess transport doesn't support maxMessageSize (infinite is always allowed). We can add support to InProcess transport, although that comes at a performance cost since today the messages aren't generally serialized. The other option is for it to just ignore the setting.

**Comment by ejona86** [comment]
Fixed by #2382
